### DISCUSSION TITLE
Anyone in the US or Canada used xDrip+ with Libre 2 sensor?

### DISCUSSION THREAD

**Discussion by kleenkanteen** [NightscoutFoundation/xDrip#2584] [Q&A] [answered]
I tried using Diabox which is getting readings from my libre 2, and then on xDrip+ set the hardware source as Nightscout Follower and the Nightscout Follow URL http://localhost:17850. But then on the xDrip+ homepage, under the graph, it says "NS entries download failed: java.net.ConnectException: Failed to connect to localhost/127.0.0.1:17850". 

edit: found the issue, the url in xDrip+ should be http://localhost:17580. works now, xDrip+ is updating with the data from Diabox

Can't download the patched LibreLink app since it only works in the EU and I am in Canada, source: https://xdrip.readthedocs.io/en/latest/install/libre2/

I am thinking of using OOP1 or OOP2 but this link (https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip) says OOP2 only works with EU sensors. And OOP1 stopped working on most phones running android 10. My phone is on android 12.

**Comment by tzachi-dar** [comment, marked as answer]
Please check connection parameters to diabox.
At least in this mail you have 17850 and 17580

**Reply by kleenkanteen** [reply]
Typo, meant to write 17850 everywhere. That was the issue, should have been 17580

**Reply by danwhill** [reply]
Sabih, I am also using the Libre 2 in Canada and have been struggling to use it in a closed loop system (AndroidAPS).  Were you successful in using diabox?  I have built it and installed it, but have not tried it on the sensor, as I'm unsure if it will simply work with an existing sensor which is already in use, or if you need to start a new sensor using the diabox app?  Is the bubble device necessary, or will diabox just work with the base libre 2 sensor?
Sorry for all the questions.

**Reply by kleenkanteen** [reply]
Yup I got it to work. You use it on an existing sensor. No bubble device necessary. You activate the sensor with the official librelink app, then you do NFC scan with diabox, and it should send the readings to diabox from them on. You can still manually scan with the librelink app perfectly fine if you want.

Btw random tip, you should leave a new libre 2 sensor unactivated for at least 24 hours before activating it. This is because your immune system attacks it as it sees it as a foreign body and around 24 hours is what it takes for the response to settle down and readings to be accurate. You can also put the libre 2 anywhere on your body where there is fat, not just under you upper arm, but also your belly, inner thighs, and even your forearms as I saw someone do.

Let me know if you get it to work

**Reply by danwhill** [reply]
It did work as you say.  I'm not sure it is automatically reading from the sensor though. It only seems to update if I do an NFC scan from the diabox app.  I do have it going to xdrip and AAPS though.  Just need the bluetooth reading to work, and it is good to go.  Did you have to do anything special for that?

**Reply by kleenkanteen** [reply]
It should pick up the readings over bluetooth from the libre 2. Make sure diabox has the proper bluetooth and always on settings. Try uninstalling diabox and try again if it doesn't work.

**Reply by danwhill** [reply]
Are you able to let me know which version of diabox you are using and have working?  When I first installed it, I would get sporadic readings for a few minutes, then none.  Now, I am not able to get any via bluetooth, only NFC.  I have re-installed various versions including the latest beta, but nothing seems to work.  Wondering if this is a Android 13 bluetooth issue since there have been a few, including the one which broke the manufacturer's LibreLink app alarms for a couple months.

**Reply by kleenkanteen** [reply]
Here is the name of the .apk: Diabox_2022_02_15_2_beta.apk 

The diabox github should have this version. latest one actually. There is a telegram channel where the developer responds, forgot how I found it.

**Reply by danwhill** [reply]
Thanks Sabih, I have tried that one.  I am beginning to wonder if it is a faulty sensor, or a bluetooth issue.  What phone are you using this on, and do you need to use xDrip+, or does AAPS still get the data from diabox without using xDrip+?  I don't imagine xdrip+ is my issue, since I tried it with and without it, and neither worked.
Out of curiosity, do your libre 2 sensors stop transmitting earlier than the sensor end date?  i.e. stop transmitting after say 9 days, but the sensor works until the 14?  That would be the situation I am in, if it were the sensor.
Thanks for your patience and help, it is much appreciated.

**Reply by kleenkanteen** [reply]
My libre 2 sensors ended on time after 14 days. Technically diabox shouldn't communicate with the libre 2, it should just silently read from it. I believe the NFC scan with diabox just grabs the identifier for the sensor to know which bluetooth reading to apply the decryption algorithm on. Just a hunch though. Initially I also had some hiccups so I understand your pain lol. I only used diabox for the duration of 2 sensors also.

I'm using a samsung galaxy s10e, android 12. I don't think AAPS can connect directly to the libre 2. Check out this other app called Juggluco in place of diabox, http://jkaltes.byethost16.com/Juggluco/mgdL/index.html.

If all else fails, consider the dexcom g6. There is a modified version of the official app that can send readings to xdrip+ reliably, I am using the dexcom right now for a clinical trial and use the modified app to send readings to xdrip+. Here is the link to the reddit post about it: https://www.reddit.com/r/dexcom/comments/y4vixk/oc_build_your_own_dexcom_app_update_now_with_g7/ (reddit is donw right now though)

**Reply by danwhill** [reply]
Finally got this working.  Had to disable most permissions and alarms on the librelink app, then turned off bluetooth, back on, scanned sensor with NFC in diabox (basically as if I was setting it up for the first time), then it started logging readings a couple hours later.  Very touchy and sporadic, but it now works.  I think there are definitely some bluetooth "issues" at play here, but it does work on the Samsung Galaxy S20 with android 13.
Thanks again for your patience, you were an immense help.
All the best

**Reply by kleenkanteen** [reply]
You are very welcome, I remember reading somewhere to turn off some permissions for the librelink app. Also had some trouble getting it to work initially, is a bit finnicky but glad to hear it works now. Cheers

**Comment by danwhill** [comment]
Thanks, that's great to hear, and very helpful.  I was afraid to try it and
ruin my sensor until it was close to ending.
I'll give it a try.

**Comment by Scoudouc** [comment]
> I tried using Diabox which is getting readings from my libre 2, and then on xDrip+ set the hardware source as Nightscout Follower and the Nightscout Follow URL http://localhost:17850. But then on the xDrip+ homepage, under the graph, it says "NS entries download failed: java.net.ConnectException: Failed to connect to localhost/127.0.0.1:17850".
> 
> edit: found the issue, the url in xDrip+ should be http://localhost:17580. works now, xDrip+ is updating with the data from Diabox
> 
> Can't download the patched LibreLink app since it only works in the EU and I am in Canada, source: https://xdrip.readthedocs.io/en/latest/install/libre2/
> 
> I am thinking of using OOP1 or OOP2 but this link (https://www.minimallooper.com/post/how-to-setup-freestyle-libre-2-and-oop2-to-use-a-native-bluetooth-connection-in-xdrip) says OOP2 only works with EU sensors. And OOP1 stopped working on most phones running android 10. My phone is on android 12.

Is this still working for you in Canada since Abbott updated the sensors?

**Comment by kleenkanteen** [comment]
What do you mean update the sensors? The libre 3? I don't know either way,
I am using Dexcom g6 now.
